modules/flowable-cmmn-engine-configurator/src/test/java/org/flowable/cmmn/test/CaseTaskTest.java

```diff
@@ -300,4 +303,37 @@ public void testCaseTaskIdVariableNameExpression() {
             processEngineRepositoryService.deleteDeployment(deployment.getId(), true);
         }
     }
+
+    @Test
+    @CmmnDeployment(resources = {"org/flowable/cmmn/test/CaseTaskTest.testCaseTask.cmmn"})
+    public void testCaseTaskWithTerminateEndEvent() {
+        Deployment deployment = processEngineRepositoryService.createDeployment()
+                .addClasspathResource("org/flowable/cmmn/test/caseTaskProcessWithTerminateEndEventOnDifferentExecution.bpmn20.xml")
+                .deploy();
+
+        try {
+            // Arrange
+            ProcessInstance processInstance = processEngineRuntimeService.startProcessInstanceByKey("terminateByEndEvent");
+            Task task = processEngineTaskService.createTaskQuery()
+                    .processInstanceId(processInstance.getProcessInstanceId())
+                    .singleResult();
+            Assumptions.assumeThat(task).isNotNull();
```

Comment:
Why are you using assumptions? If for whatever reason the task is null the test would be ignored since the assumption is not met, it won't actually fail

Reply:
To ensure that the test is only failing in cases the test is supposed to fail. The test is not supposed to test if a simple user task is created, the only purpose of this test is to test that in case somebody triggers a terminate end event, the case is terminated.